corehq/apps/app_manager/util.py

```diff
@@ -134,9 +134,11 @@ def change_xmlns(xform, old_xmlns, new_xmlns):
     else:
         GENERIC_XMLNS = "http://www.w3.org/2002/xforms"
         tag_xmlns = xform.data_node.tag_xmlns
+        form_xmlns = "http://openrosa.org/formdesigner/%s" % form.get_unique_id()
         if not tag_xmlns or tag_xmlns == GENERIC_XMLNS:  # no xmlns
-            xmlns = "http://openrosa.org/formdesigner/%s" % form.get_unique_id()
-            xml = change_xmlns(xform, GENERIC_XMLNS, xmlns)
+            xml = change_xmlns(xform, GENERIC_XMLNS, form_xmlns)
+        elif tag_xmlns != form_xmlns:
+            xml = change_xmlns(xform, tag_xmlns, form_xmlns)
```

Comment:
@dannyroberts, @snopoke or anyone in @dimagi/team-commcare-hq do you know if this will cause problems? Also see https://github.com/dimagi/commcare-hq/pull/17250 for more context.